solidity-checks/src/main/java/org/sonarsource/solidity/checks/CheckVerifier.java

```diff
@@ -56,13 +55,33 @@ public static void verifyNoIssue(IssuableVisitor check, String relativePath) {
     new CheckVerifier(check, relativePath, false);
   }
 
+  public static boolean verifyIssueOnFile(IssuableVisitor checkVisitor, String relativePath) {
+    File file = new File(relativePath); ///// TODO fix architecture
+    SingleFileVerifier verifier = SingleFileVerifier.create(file.toPath(), UTF_8);
+    CharStream cs;
+    try {
+      cs = CharStreams.fromFileName(relativePath);
+      SolidityParser parser = Utils.returnParserFromParsedFile(cs);
+      TestRuleContext testRuleContext = new TestRuleContext(verifier);
+      checkVisitor.setRuleContext(testRuleContext);
+      checkVisitor.visit(parser.sourceUnit());
+      if (testRuleContext.issueOnFile) {
+        return true;
+      }
+    } catch (IOException e) {
+      LOG.debug(e.getMessage(), e);
+    }
+    return false;
+  }
+
   /*
    * class used to report for unit tests
    */
 
   private static class TestRuleContext implements RuleContext {
 
     SingleFileVerifier verifier;
+    public boolean issueOnFile = false;
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Make issueOnFile a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=33&issues=AWRuvWkXI66nRh8YCYJk&open=AWRuvWkXI66nRh8YCYJk)

[ISSUE_KEY]: AWRuvWkXI66nRh8YCYJk